ComputeExamples.md

```diff
@@ -0,0 +1,676 @@
+
+## 1. Initialize services
+
+Ocean.js uses off-chain services for metadata (Aquarius) and consuming datasets (Provider).
+
+We start by initializing the services. To do this, we clone the Barge repository and run it. This will run the current default versions of [Aquarius](https://github.com/oceanprotocol/aquarius), [Provider](https://github.com/oceanprotocol/provider), and [Ganache](https://github.com/trufflesuite/ganache) with [our contracts](https://github.com/oceanprotocol/contracts) deployed to it.
+
+```bash
+git clone https://github.com/oceanprotocol/barge.git
+cd barge/
+./start_ocean.sh --with-provider2 --no-dashboard --with-c2d
+```
+
+## 2. Create a new node.js project with Typescript
+
+Start by creating a new Node.js project. Open a new terminal and enter the following commands:
+
+```bash
+mkdir compute-quickstart
+cd compute-quickstart
+npm init
+# Answer the questions in the command line prompt
+touch compute.ts
+# On linux press CTRL + D to save
+```
+
+Next, we need to setup our typescript compiler options. Create a new file called `tsconfig.json` in the root of the `compute-quickstart` directory. copy the following content into the file:
+
+```json
+{
+  "compilerOptions": {
+    "resolveJsonModule": true,
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "lib": [
+      "ESNext",
+      "dom"
+    ],
+    "declaration": true,
+    "module": "ESNext",
+    "target": "ESNext",
+    "removeComments": false,
+    "experimentalDecorators": true,
+    "preserveConstEnums": true,
+    "outDir": "./",
+    "sourceMap": true,
+    "declarationDir": "./d"
+  },
+  "include": [
+    "compute.ts"
+  ]
+}
+```
+
+Now you can compile your typescript with the following command:
+
+```bash
+tsc
+```
+
+To run your script as we go along, you can use the following command:
+
+```bash
+node compute.js
+```
+
+## 3. Install dependencies
+
+Install dependencies running the following command in your terminal:
+
+```bash
+npm install @oceanprotocol/lib crypto-js web3 web3-utils typescript @types/node ts-node
+```
+
+## 4. Import dependencies and add variables, constants and helper methods
+
+Now open the `compute.ts` file in your text editor.
+
+### 4.1. Dependencies
+
+Start by importing all of the necessary dependencies
+
+```Typescript
+
+import { SHA256 } from 'crypto-js'
+import { AbiItem } from 'web3-utils'
+import {
+  ProviderInstance,
+  Aquarius,
+  NftFactory,
+  Datatoken,
+  Nft,
+  ZERO_ADDRESS,
+  transfer,
+  sleep,
+  approveWei,
+  ProviderComputeInitialize,
+  ConsumeMarketFee,
+  ComputeAlgorithm,
+  ComputeAsset,
+  Config,
+  Files,
+  DDO,
+  NftCreateData,
+  DatatokenCreateParams,
+  calculateEstimatedGas,
+  sendTx
+} from '@oceanprotocol/lib'
+import { getAddresses, getTestConfig, web3 } from '@oceanprotocol/lib/dist/test/config'
```

Comment:
this would be another source of error, we should not instruct people to get methods from our test configs. And NEVER should we instruct people to get web3 from there

Reply:
sure will remove this, and have the logic in the file to be more understandable for the users